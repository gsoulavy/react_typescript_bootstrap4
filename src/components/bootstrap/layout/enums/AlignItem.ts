enum AlignItem {
    Start = "align-items-start",
    Center = "align-items-center",
    End = "align-items-end"
}

export default AlignItem;