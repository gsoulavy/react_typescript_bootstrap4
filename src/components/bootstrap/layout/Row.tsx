import * as React from "react";
import * as helper from "./../helpers/StatelessHelper";
import ComponentProps from "./../interfaces/IComponentProps";
import AlignItem from "./enums/AlignItem";

interface Props extends ComponentProps {
    align?: AlignItem
}

const Row : React.StatelessComponent<Props> = (prop) => {
    return (
        <div className={`row${helper.mergeClassName(prop.align)}${helper.mergeClassName(prop.applyClass)}`.trim()}>
            {prop.children}
        </div>
    );
};

export default Row;