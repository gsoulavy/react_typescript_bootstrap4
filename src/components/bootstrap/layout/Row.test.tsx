import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Row from "./Row";
import AlignItem from "./enums/AlignItem";

describe("<Row />", () => {

    it("no params -> class='row'", () => {
        const component = shallow(<Row >hi</Row>)
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    });

    it("applyClass='main' -> class='row main'", () => {
        const component = shallow(<Row applyClass='main'>hi</Row>)
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    });

    it("align='Start' -> class='row align-item-start'", () => {
        const component = shallow(<Row align={ AlignItem.Start }>hi</Row>)
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    });

    it("applyClass='main' + align='Start' -> class='row align-item-start'", () => {
        const component = shallow(<Row applyClass="main" align={ AlignItem.Start }>hi</Row>)
        const tree = toJson(component);
        expect(tree).toMatchSnapshot();
    });

});

