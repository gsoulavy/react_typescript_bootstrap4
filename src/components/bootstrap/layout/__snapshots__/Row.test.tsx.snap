// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Row /> align='Start' -> class='row align-item-start' 1`] = `
<div
  className="row align-items-start"
>
  hi
</div>
`;

exports[`<Row /> applyClass='main' + align='Start' -> class='row align-item-start' 1`] = `
<div
  className="row align-items-start main"
>
  hi
</div>
`;

exports[`<Row /> applyClass='main' -> class='row main' 1`] = `
<div
  className="row main"
>
  hi
</div>
`;

exports[`<Row /> no params -> class='row' 1`] = `
<div
  className="row"
>
  hi
</div>
`;
